lib/api/route/channels.dart

```diff
@@ -38,3 +39,30 @@ class GetStreamTopicsEntry {
 
   Map<String, dynamic> toJson() => _$GetStreamTopicsEntryToJson(this);
 }
+
+/// https://zulip.com/api/update-user-topic
+/// https://zulip.com/api/mute-topic (legacy)
+Future<void> updateUserTopic(ApiConnection connection, {
```

Comment:
Let's keep the API binding thin: the function named `updateUserTopic` just invokes the endpoint `update-user-topic`, not another endpoint.

To handle this compatibility fallback, we can use the same structure as `getMessage` and its wrapper `getMessageCompat`.